App/In-App Actions/URLMenuPresenter.swift

```diff
@@ -292,6 +294,72 @@ private func chromifyURL(_ url: URL) -> URL {
     return components.url!
 }
 
+func downloadVideo(with url: URL, completion: @escaping (URL?) -> ()) {
+    URLSession.shared.downloadTask(with: url) { url, response, error in
+        guard let tempUrl = url, error == nil else {
+            return completion(nil)
+        }
+  
+        let fileManager = FileManager.default
+        let documentsUrl = fileManager.urls(for: .documentDirectory, in: .userDomainMask).first!
+        let videoFileUrl = documentsUrl.appendingPathComponent(response?.suggestedFilename ?? "temp.mp4")
+        
+        if fileManager.fileExists(atPath: videoFileUrl.path) {
+            try? fileManager.removeItem(at: videoFileUrl)
+        }
+        do {
+            try fileManager.moveItem(at: tempUrl, to: videoFileUrl)
+            Log.d("url: \(videoFileUrl)")
+            completion(videoFileUrl)
+        } catch {
+            completion(nil)
+        }
+    }.resume()
+}
+
+func saveToPhotos(_ url: URL, overlay: MRProgressOverlayView?, completion: @escaping (ErrorWorkaround?) -> ()) {
+    PHPhotoLibrary.requestAuthorization { status in
```

Comment:
This is a better place for this and it's good that we wait for the answer before continuing. I wonder if an even better place is before we start downloading? Then we're not using up bandwidth just for the user to say "nah". Like the sequence would be:

1. Action sheet "Save Video".
2. Request authorization.
3. Download file.
4. Add file to photo library.

Reply:
Went ahead and did this.